src/command_modules/azure-cli-interactive/HISTORY.rst

```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+0.3.17
+++++++
+* Persist history across different sessions
+* History while in non-default scope disappears after scope is changed and is not saved.
```

Comment:
I don't really understand this. If I was in a scope, say "network" and ran `public-ip create ...` why would `network public-ip- create ...` not appear in history?  It sounds like `public-ip create` would appear while I'm in the "network" scope but then when I change scope, that history vanishes.  Is that right?

Reply:
Yes, that's right, the reason why I did this is because the scoping commands `%% <whatever>` would not apply once the scope is changed, so at the very least, they should not be saved.

The other option would be to match for different scenarios and save accordingly. i.e., scope commands are not saved, cli commands have the scope appended, non-azure cli commands (using `#`) are saved alone, etc.
In addition, inside a scope, some history is no longer valid. i.e. all commands from other scopes. Thus, we would also need to parse through the history if all scopes use the same source.
This can be confusing, a user may wonder why some things are showing up, some aren't, and why some differ from what they typed.

To me, having a simpler rule defined, where scoping creates a different "environment" of sorts with its own history seems much more intuitive. Once that environment is left, the history is cleaned up and gone.

Reply:
After discussion with @tjprescott, decided to go with one source of history for all commands, no matter the scope. 
It will be up to the user to deal with history from different scopes.

Reply:
After our discussion, I think what we agreed upon was:
1) not deleting the history regardless of scope
2) just presenting the text history, regardless of scope

IMO its a little less user-focused, but it's better than being broken.  If you add support for ~ then bonus points :)